### PR TITLE
Migrated code from old repository

### LINKED ISSUE #6: Initialize Compilation service

### PR DESCRIPTION
**Pull request by AHGIJMKLKKZNPJKQR**
resolves #6

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+FROM rust:1.68-slim-buster as builder
+WORKDIR /app/src
+
+# Force crates.io init for better docker caching
+RUN cargo search --limit 0
```
Comment (ghost) [inline Dockerfile]: Add the caching stuff as it was
```suggestion
# Force crates.io init for better docker caching
COPY docker/caching.rs src/main.rs
COPY Cargo.lock .
COPY Cargo.toml .
RUN cargo build --release
```
Where `docker/caching.rs` is:
```
// DO NOT CHANGE!
// Helps to improve docker build caching
// https://stackoverflow.com/questions/58473606/cache-rust-dependencies-with-docker-build

fn main() {}
```

File: swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,113 @@
+openapi: 3.0.3
+info:
+  title: Agartex Compilation Service
+  description: |-
+    This is a server for compiling Latex documents
+  version: 0.0.1
+servers:
+  - url: http://localhost:3002
```
Comment (ghost) [inline swagger.yaml]: Port

File: swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+tags:
+  - name: compile
+    description: LaTeX document compilation
+paths:
+  /projects/{projectId}/pdf:
+    post:
+      security:
+        - session_id: []
```
Comment (ghost) [inline swagger.yaml]: `session_id` will be no more here, probably `user_id` will be instead.

File: config.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+    "server_addr": "0.0.0.0:3300",
+    "tmp_file_dir": "/tmp/agar_service/",
+    "latexmk_path": "latexmk"
+}
```
Comment (ghost) [inline config.json:5]: Can we override those by setting env variables?

**Review by ghost — CHANGES_REQUESTED** [review]

File: config.json (hunk 4 above)
Comment (AHGIJMKLKKZNPJKQR) [reply config.json:5]: Not currently, I can make it so that's a thing.

**Review by ghost — APPROVED** [review]

**Pull request merged**